File: app/models/person.py
from app.models.enum import enum


class Person:

    def __init__(self, surname, first_name, date_of_birth, gender):
        self.id = None
        self.surname = surname
        self.first_name = first_name
        self.date_of_birth = date_of_birth
        self.gender = gender

    def jsonify(self):
        return {
            u"id": unicode(self.id),
            u"surname": unicode(self.surname),
            u"first_name": unicode(self.first_name),
            u"date_of_birth": unicode(self.date_of_birth),
            u"gender": unicode(self.gender)
        }

gender = enum(MALE="male", FEMALE="female")

File: application/client/models/person.js
var superAgent = require('superagent-promise');
var endpoint = require('../config/api-config.js').PEOPLE_ENDPOINT;

var Person = function(surname, firstName, dateOfBirth, gender) {
    this.surname = surname;
    this.firstName = firstName;
    this.dateOfBirth = dateOfBirth;
    this.gender = gender;
};

Person.GENDER = {
    FEMALE: 'female',
    MALE: 'male'
};

Person.prototype.save = function() {
    return addPerson({
        surname: this.surname,
        first_name: this.firstName,
        date_of_birth: this.dateOfBirth,
        gender: this.gender
    });
};

Person.build = function(data) {
    var person = new Person(data.surname, data.first_name);
    person.id = data.id;
    return person;
};

var addPerson = function(data) {
    return superAgent.post(endpoint, data).end().then(function(response) {
        return Person.build(response.body);
    });
};

module.exports = Person;
